src/sinks/util/metrics_buffer.rs

```diff
@@ -0,0 +1,488 @@
+use super::batch::Batch;
```

Comment:
we should probably create a `sinks/util/buffer/mod.rs` and add metrics and partiton into here. I think this would make sense? @lukesteensen